### ISSUE TITLE
Don't prefix injectGlobal with a selector

### ISSUE THREAD

**Issue by Gautierrr** [styled-components/stylelint-processor-styled-components#26]
This code...
```js
injectGlobal`
  html {
    color: blue;
  }
`;
```

... with `selector-max-compound-selectors` rule set to `1` returns this validation issue
```sh
Expected ".selector html" to have no more than 1 compound selectors
```

**Comment by mxstbr** [comment]
Ah crap, need to add a special case for `injectGlobal`!

**Comment by emilgoldsmith** [comment]
@mxstbr I'm just trying to kill off the last few bugs before v1.0.

For this one are you thinking that since `injectGlobal` is semantically meant to be in global scope anyway we don't need to add a selector? Which makes sense.

Because I guess @Gautierrr's problem would also be relevant if we for example did something like this:
```js
const Button = styled.div`
  span {
    background-color: red;
  }
`;
```
right?

Since we currently consistently wrap everything in selectors a fix could also be just upping your rule here + 1, right? As in if you in effect want it to be 1, you could make it 2?

**Comment by mxstbr** [comment]
> Since we currently consistently wrap everything in selectors a fix could also be just upping your rule here + 1, right?

If you're doing `styled.x` with a selector you have to bump the rule, that makes sense. But the `injectGlobal` bit shouldn't throw an error since you have to use a selector there, so we only need a special case for `injectGlobal`.

**Comment by emilgoldsmith** [comment]
> If you're doing styled.x with a selector you have to bump the rule, that makes sense. But the injectGlobal bit shouldn't throw an error since you have to use a selector there, so we only need a special case for injectGlobal.

Hmm, yeah I see the logic of having to use selectors there, but I guess the error is still just as weird in a `styled.x` call as in an `injectGlobal` call as one intuitively thinks that there shouldn't be any problems with either of them. 
I think I'd vote for consistency, no? Imagine someone using the rule and everything working fine in `injectGlobal` and then they use a selector in `styled.x` and it throws an error and they're confused?

Okay I'm gonna do a stream of consciousness thing here as I just thought of a possible counterargument :P.
Is your argument (or the argument) that since this specific rule about max compound selectors set at for example 1 is actually supposed to throw at
```js
const Button = styled.div`
  span {
    background-color: red;
  }
`;
```
because the css equivalent would be:
```css
.some-classname-or-other-selector {
  span {
    background-color: red;
  }
}
```
?
Because then I actually suddenly see the argument as then we'd be currently correctly enforcing this rule in `styled.x` while `injectGlobal` would exactly be better off without.

**Comment by emilgoldsmith** [comment]
Also, should we change anything about the way we handle the `css` helper?

**Comment by mxstbr** [comment]
So, `injectGlobal` shouldn't throw because

```JS
injectGlobal`
  body {
    color: blue;
  }
`
```

injects

```CSS
body {
  color: blue;
}
```

Which is perfectly fine! But when using `styled.x` or `css`

```JS
const Button = styled.button`
  span {
    color: blue;
  }
`

const fragment = css`
  span {
    color: blue;
  }
`

const Comp = styled.div`${fragment}`
```

injects

```CSS
.Button-asdf123 span {
  color: blue;
}
```

which is just as much an anti-pattern as having nested selectors, so that should throw.

**Comment by emilgoldsmith** [comment]
Okay, I'm thoroughly convinced :). I'll submit a quick PR later today with the changes to injectGlobal, shouldn't be many lines of code that need changing.
